### PR TITLE
Trim dependencies

### PR DESCRIPTION
**Pull request by c410-f3r**
All tests are passing.

`cargo check --all-features --tests`, from 589 to 566 dependencies.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -41,11 +41,11 @@ serde_cbor = { version = "0.11.2"}
 uuid = { version = "1.3", features = ["v4", "serde"] }
 sys-info = "0.9.1"
 
-config = "~0.13.3"
+config = { default-features = false, features = ["yaml"], version = "~0.13.3" }
```
Comment (c410-f3r) [inline Cargo.toml:44]: Could not find other configs that aren't `yaml`. See https://github.com/mehcode/config-rs/blob/master/Cargo.toml#L18

**Comment by generall** [comment]
hi @c410-f3r, could you please rebase your PR to `dev` branch?

**Comment by c410-f3r** [comment]
> hi @c410-f3r, could you please rebase your PR to `dev` branch?

Done

**Review by Shevilll — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (timvisee) [reply Cargo.toml:45]: We do load/store some JSON structures throughout Qdrant, but I don't think they ever reach `Config`.

Thanks for tagging this.

**Comment by timvisee** [comment]
@c410-f3r We have a few other `Cargo.toml`s. Are you interested in looking at those as well?

https://github.com/search?q=repo%3Aqdrant%2Fqdrant++Cargo.toml&type=code

**Comment by Govthub** [comment]
Am here to fixed your issue, drop your telegram or WhatsApp number so we can handle it

File: Cargo.toml [hunk 2]
```diff
@@ -41,11 +41,11 @@ serde_cbor = { version = "0.11.2"}
...
 
 tokio = { version = "~1.27", features = ["full"] }
 
-actix-web = { version = "4.3.1", optional = true, features = ["openssl"] }
+actix-web = { default-features = false, features = ["compress-zstd", "macros", "openssl"], optional = true, version = "4.3.1" }
```
[the remark this answers is not in the document]
Comment (c410-f3r) [reply Cargo.toml:49]: Changed to `compress-gzip`

**Comment by c410-f3r** [comment]
> @c410-f3r We have a few other `Cargo.toml`s. Are you interested in looking at those as well?
> 
> https://github.com/search?q=repo%3Aqdrant%2Fqdrant++Cargo.toml&type=code

Sure.

It is not very trivial to catch these unnecessary dependencies without breaking the existing code so I prefer to investigate more in following PRs.

**Comment by timvisee** [comment]
> Sure.
> 
> It is not very trivial to catch these unnecessary dependencies without breaking the existing code so I prefer to investigate more in following PRs.

Of course. Thanks a lot for taking time to do this!

We'll make sure to merge this.

**Review by timvisee — APPROVED** [review]

**Comment by c410-f3r** [comment]
ping

**Pull request closed without merging**